app/models/product.rb

```diff
@@ -15,6 +15,9 @@ class Product < ActiveRecord::Base
   has_many   :price_policies
   has_many   :training_requests, dependent: :destroy
 
+  cattr_accessor(:types) { ConfigurableArray.new([Instrument, Item, Service, Bundle]) }
```

Comment:
I could be convinced to go back to the old way where we did the `index` and `insert(index, xxx)` within the engines.